process/interceptors/processor/equivalentProofsInterceptorProcessor.go

```diff
@@ -40,14 +48,38 @@ func checkArgsEquivalentProofs(args ArgEquivalentProofsInterceptorProcessor) err
 	if check.IfNil(args.Marshaller) {
 		return process.ErrNilMarshalizer
 	}
+	if check.IfNil(args.PeerShardMapper) {
+		return process.ErrNilPeerShardMapper
+	}
+	if check.IfNil(args.NodesCoordinator) {
+		return process.ErrNilNodesCoordinator
+	}
 
 	return nil
 }
 
-// Validate checks if the intercepted data can be processed
-// returns nil as proper validity checks are done at intercepted data level
-func (epip *equivalentProofsInterceptorProcessor) Validate(_ process.InterceptedData, _ core.PeerID) error {
-	return nil
+// Validate checks if the intercepted data was sent by an eligible node
+func (epip *equivalentProofsInterceptorProcessor) Validate(data process.InterceptedData, pid core.PeerID) error {
+	interceptedProof, ok := data.(interceptedEquivalentProof)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	proof := interceptedProof.GetProof()
+
+	eligibleList, err := epip.nodesCoordinator.GetAllEligibleValidatorsPublicKeysForShard(proof.GetHeaderEpoch(), proof.GetHeaderShardId())
```

Comment:
Is `GetAllEligibleValidatorsPublicKeysForShard` a "fast-enough" operation?

Reply:
it is taken from nodes config, which is in memory, in a map in nodesCoordinator, so it should be fast-enough

Reply:
I think we can keep the check for eligible nodes, but cache it in a map inside the interceptor, so for the entire epoch we only do the computation once, and then just access the map.

Reply:
pushed the suggestion